File: samples_java/src/main/java/MetaSample.java
import com.aliyun.cloudcontrol20220606.Client;
import com.aliyun.cloudcontrol20220606.models.*;
import com.aliyun.tea.TeaException;
import com.aliyun.teaopenapi.models.Config;
import com.aliyun.teautil.models.RuntimeOptions;
import com.google.gson.Gson;
import java.util.HashMap;
import java.util.Map;

/**
 * @author bzran
 * @description 查询资源meta
 */
public class MetaSample {

    /**
     * 云厂商
     */
    private final static String provider = "Aliyun";

    /**
     * 云产品
     */
    private final static String productCode = "ECS";

    /**
     * 资源类型
     */
    private final static String resourceTypeCode = "Instance";

    /**
     * SDK Client
     */
    private static com.aliyun.cloudcontrol20220606.Client cloudControlClient;

    public static void main(String[] args) {

        try {
            Config config = new Config()
                    // 您的AccessKey ID
                    .setAccessKeyId("your_ak")
                    // 您的AccessKey Secret
                    .setAccessKeySecret("your_sk")
                    //设置sdk超时时间
                    .setReadTimeout(20000);
            config.endpoint = "cloudcontrol.aliyuncs.com";
            cloudControlClient = new Client(config);

            //列举产品列表
            String products = listProducts();
            //列举ECS下的资源列表
            String resourceTypes = listResourceTypes(productCode);
            //查询ECS的Instance资源meta
            String resourceType = getResourceType(productCode, resourceTypeCode);
            //查询ECS的Instance资源meta(国际化，返回英文描述)
            String resourceTypeEn = getResourceTypeEn(productCode, resourceTypeCode);
            //查询ECS的Instance资源支持的Region
            String regions = listDataSources(productCode, resourceTypeCode, "RegionId");
        } catch (TeaException e) {
            // 如有需要，请打印 error
            System.out.println("Error code: " + e.getCode());
            System.out.println("Error message: " + e.getMessage());
        } catch (Exception e) {
            TeaException error = new TeaException(e.getMessage(), e);
            // 打印 error
            System.out.println(com.aliyun.teautil.Common.assertAsString(error.message));
        }
    }

    /**
     * 列举产品列表
     */
    private static String listProducts() throws Exception {
        ListProductsRequest listProductsRequest = new ListProductsRequest();
        ListProductsResponse listProductsResponse = cloudControlClient.listProducts(provider, listProductsRequest);
        return new Gson().toJson(listProductsResponse.getBody().getProducts());
    }

    /**
     * 列举产品下的资源列表
     */
    private static String listResourceTypes(String product) throws Exception {
        ListResourceTypesRequest listResourceTypesRequest = new ListResourceTypesRequest();
        ListResourceTypesResponse listResourceTypesResponse = cloudControlClient.listResourceTypes(provider, product, listResourceTypesRequest);
        return new Gson().toJson(listResourceTypesResponse.getBody().getResourceTypes());
    }

    /**
     * 查询资源meta详情
     */
    private static String getResourceType(String product, String resourceType) throws Exception {
        GetResourceTypeRequest getResourceTypeRequest = new GetResourceTypeRequest();
        GetResourceTypeResponse getResourceTypeResponse = cloudControlClient.getResourceType(provider, product, resourceType, getResourceTypeRequest);
        return new Gson().toJson(getResourceTypeResponse.getBody().getResourceType());
    }

    /**
     * 查询资源meta详情(国际化英文版)
     */
    private static String getResourceTypeEn(String product, String resourceType) throws Exception {
        Map<String, String> headers = new HashMap<>();
        headers.put("x-acs-accept-language", "en_US");
        RuntimeOptions runtime = new RuntimeOptions();
        GetResourceTypeRequest getResourceTypeRequest = new GetResourceTypeRequest();
        GetResourceTypeResponse getResourceTypeResponse = cloudControlClient.getResourceTypeWithOptions(provider, product, resourceType, getResourceTypeRequest, headers, runtime);
        return new Gson().toJson(getResourceTypeResponse.getBody().getResourceType());
    }

    /**
     * 查询datasource(支持地域、可用区等)
     */
    private static String listDataSources(String product, String resourceType, String attributeName) throws Exception {
        ListDataSourcesRequest request = new ListDataSourcesRequest();
        ListDataSourcesResponse listDataSourcesResponse = cloudControlClient.listDataSources(provider, product, resourceType, attributeName, request);
        return new Gson().toJson(listDataSourcesResponse.getBody().getDataSources());
    }

}

File: samples_python/meta_sample.py
from alibabacloud_cloudcontrol20220606.client import Client as CloudControlClient
from alibabacloud_tea_openapi import models as open_api_models
from alibabacloud_cloudcontrol20220606 import models as cloud_control_models
from alibabacloud_tea_util import models as util_models
from alibabacloud_tea_util.client import Client as UtilClient


class MetaSample:

    def __init__(self):
        pass

    @staticmethod
    def create_client(
    ) -> CloudControlClient:
        """
        使用AK&SK初始化账号Client
        @return: Client
        @throws Exception
        """
        config = open_api_models.Config(
            # 您的AccessKey ID(改为自己的)
            access_key_id="your_ak",
            # 您的AccessKey Secret(改为自己的)
            access_key_secret="your_sk",
            # 公司UserAgent信息(改为自己的)
            user_agent="vendor/AnyCloudCompany-CloudManager@2.1.0",
            # 设置sdk读取超时时间
            read_timeout=20000
        )
        # 访问的域名
        config.endpoint = 'cloudcontrol.aliyuncs.com'
        return CloudControlClient(config)

    @staticmethod
    def list_products(
    ) -> list:
        """
        列举产品列表
        @return: list
        @throws Exception
        """
        list_products_request = cloud_control_models.ListProductsRequest()
        list_products_response = cloud_control_client.list_products(provider, list_products_request)
        if list_products_response.status_code == 200:
            return list_products_response.body.products
        else:
            raise Exception("Invalid status_code!")

    @staticmethod
    def list_resource_types(
        product: str
    ) -> list:
        """
        列举产品下的资源列表
        @return: list
        @throws Exception
        """
        list_resource_types_request = cloud_control_models.ListResourceTypesRequest()
        list_resource_types_response = cloud_control_client.list_resource_types(provider, product,
                                                                                list_resource_types_request)
        if list_resource_types_response.status_code == 200:
            return list_resource_types_response.body.resource_types
        else:
            raise Exception("Invalid status_code!")

    @staticmethod
    def get_resource_type(
        product: str,
        resource_type: str
    ) -> cloud_control_models.GetResourceTypeResponseBodyResourceType:
        """
        查询资源meta详情
        @return: GetResourceTypeResponseBodyResourceType
        @throws Exception
        """
        get_resource_type_request = cloud_control_models.GetResourceTypeRequest()
        get_resource_type_response = cloud_control_client.get_resource_type(provider, product, resource_type,
                                                                            get_resource_type_request)
        if get_resource_type_response.status_code == 200:
            return get_resource_type_response.body.resource_type
        else:
            raise Exception("Invalid status_code!")

    @staticmethod
    def get_resource_type_en(
        product: str,
        resource_type: str
    ) -> cloud_control_models.GetResourceTypeResponseBodyResourceType:
        """
        查询资源meta详情(国际化英文版)
        @return: str
        @throws Exception
        """
        get_resource_type_request = cloud_control_models.GetResourceTypeRequest()
        runtime = util_models.RuntimeOptions()
        headers = {'x-acs-accept-language': 'en_US'}
        get_resource_type_response = cloud_control_client.get_resource_type_with_options(
            provider, product, resource_type, get_resource_type_request, headers, runtime)
        if get_resource_type_response.status_code == 200:
            return get_resource_type_response.body.resource_type
        else:
            raise Exception("Invalid status_code!")

    @staticmethod
    def list_data_source(
        product: str,
        resource_type: str,
        attribute_name: str
    ) -> list:
        """
        查询datasource(支持地域、可用区等)
        @return: list
        @throws Exception
        """
        list_data_source_request = cloud_control_models.ListDataSourcesRequest()
        list_data_source_response = cloud_control_client.list_data_sources(provider, product, resource_type,
                                                                           attribute_name, list_data_source_request)
        if list_data_source_response.status_code == 200:
            return list_data_source_response.body.data_sources
        else:
            raise Exception("Invalid status_code!")


if __name__ == '__main__':
    # 云厂商
    provider = "Aliyun"
    # 产品code
    product = "ECS"
    # 资源code
    resourceType = "Instance"

    try:
        # 初始化sdk
        cloud_control_client = MetaSample.create_client()
        # 列举产品列表
        products = MetaSample.list_products()
        # 列举ECS下的资源列表
        resource_types = MetaSample.list_resource_types(product)
        # 查询ECS的Instance资源meta
        resource_type = MetaSample.get_resource_type(product, resourceType)
        # 查询ECS的Instance资源meta(国际化，返回英文描述)
        resource_type_en = MetaSample.get_resource_type_en(product, resourceType)
        # 查询ECS的Instance资源支持的Region
        data_sources = MetaSample.list_data_source(product, resourceType, 'RegionId')
        a =0
    except Exception as error:
        # 打印 error
        print(UtilClient.assert_as_string(error.message))
